src/Middleware/OutputCaching/src/OutputCacheKeyProvider.cs

```diff
@@ -58,11 +64,43 @@ public string CreateStorageKey(OutputCacheContext context)
         }
     }
 
+    public static bool ContainsDelimiters(string? value)
+    {
+        return !string.IsNullOrEmpty(value) && value.IndexOfAny(KeyDelimiters) > -1;
+    }
+
+    public static bool TryAppendKeyPrefix(OutputCacheContext context, StringBuilder builder)
+    {
+        var cacheKeyPrefix = context.CacheVaryByRules.CacheKeyPrefix;
+
+        if (!string.IsNullOrEmpty(cacheKeyPrefix))
+        {
+            if (ContainsDelimiters(cacheKeyPrefix))
+            {
+                return false;
+            }
+
+            builder
+                .Append(context.CacheVaryByRules.CacheKeyPrefix)
+                .Append(KeyDelimiter);
+        }
+
+        return true;
+    }
+
     // GET<delimiter>SCHEME<delimiter>HOST:PORT/PATHBASE/PATH
-    public void AppendBaseKey(OutputCacheContext context, StringBuilder builder)
+    public bool TryAppendBaseKey(OutputCacheContext context, StringBuilder builder)
     {
         var request = context.HttpContext.Request;
 
+        if (ContainsDelimiters(request.Method) ||
+            ContainsDelimiters(request.Scheme) ||
```

Comment:
Method and scheme can't have control characters. Path might after un-escaping (and Value is un-escaped)

Reply:
Since I was able to set it via code, I assume I would still check for it.